patching/src/main/java/org/jboss/as/patching/management/LocalPatchRollbackHandler.java

```diff
@@ -34,38 +34,38 @@
 import org.jboss.as.patching.tool.PatchTool;
 import org.jboss.as.patching.tool.PatchingResult;
 import org.jboss.dmr.ModelNode;
-import org.jboss.msc.service.ServiceRegistry;
 
 /**
  * @author Emanuel Muckenhuber
  */
-public class LocalPatchRollbackHandler implements OperationStepHandler {
+public class LocalPatchRollbackHandler extends PatchStreamResourceOperationStepHandler {
 
     public static final LocalPatchRollbackHandler INSTANCE = new LocalPatchRollbackHandler();
 
     @Override
-    public void execute(final OperationContext context, final ModelNode operation) throws OperationFailedException {
-        final String patchId = PatchResourceDefinition.PATCH_ID.resolveModelAttribute(context, operation).asString();
-        final boolean rollbackTo = PatchResourceDefinition.ROLLBACK_TO.resolveModelAttribute(context, operation).asBoolean();
-        final boolean restoreConfiguration = PatchResourceDefinition.RESET_CONFIGURATION.resolveModelAttribute(context, operation).asBoolean();
-
-        // Acquire the lock and check the write permissions for this operation
-        final ServiceRegistry registry = context.getServiceRegistry(true);
-        final InstallationManager installationManager = (InstallationManager) registry.getRequiredService(InstallationManagerService.NAME).getValue();
+    protected void execute(final OperationContext context, final ModelNode operation, final InstallationManager installationManager, final String patchStream) throws OperationFailedException {
 
         if (installationManager.requiresRestart()) {
             throw PatchLogger.ROOT_LOGGER.serverRequiresRestart();
         }
 
+        final ModelNode patchIdNode = PatchResourceDefinition.PATCH_ID.resolveModelAttribute(context, operation);
+        if(!patchIdNode.isDefined()) {
+            throw ControllerLogger.ROOT_LOGGER.nullNotAllowed(PatchResourceDefinition.PATCH_ID.getName());
+        }
```

Comment:
This doesn't seem right. If undefined is not allowed the AttributeDefinition should be configured that way and L52 will throw the exception, removing the need for this if block. But PatchResourceDefinition.PATCH_ID has allowNull set to 'true' which means the operation description will incorrectly say undefined is ok.

Perhaps the same AttributeDefinition is being used for two purposes?